cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -286,25 +293,12 @@ protected void startUp() throws Exception {
         Collections.sort(jarFiles);
         classpath.addAll(jarFiles);
         classpath.add("job.jar/classes");
-        String applicationClasspath
-          = Joiner.on(",").join(MapReduceContainerHelper.getMapReduceClassPath(mapredConf, classpath));
-
-        // Generate and localize the launcher jar to control the classloader of MapReduce containers processes
-        Location launcherJar = createLauncherJar(applicationClasspath, tempLocation);
-        job.addCacheFile(launcherJar.toURI());
 
-        // The only thing in the container classpath is the launcher.jar
-        // The MapReduceContainerLauncher inside the launcher.jar will creates a MapReduceClassLoader and launch
-        // the actual MapReduce AM/Task from that
-        // We explicitly localize the mr-framwork, but not use it with the classpath
-        URI frameworkURI = MapReduceContainerHelper.getFrameworkURI(mapredConf);
-        if (frameworkURI != null) {
-          job.addCacheArchive(frameworkURI);
-        }
+        // Add the mapreduce application classpath at last
+        MapReduceContainerHelper.getMapReduceClassPath(mapredConf, classpath);
 
-        mapredConf.unset(MRJobConfig.MAPREDUCE_APPLICATION_FRAMEWORK_PATH);
-        mapredConf.set(MRJobConfig.MAPREDUCE_APPLICATION_CLASSPATH, launcherJar.getName());
-        mapredConf.set(YarnConfiguration.YARN_APPLICATION_CLASSPATH, launcherJar.getName());
+        mapredConf.set(MRJobConfig.MAPREDUCE_APPLICATION_CLASSPATH, Joiner.on(",").join(classpath));
+        mapredConf.set(YarnConfiguration.YARN_APPLICATION_CLASSPATH, Joiner.on(",").join(classpath));
```

Comment:
Because depending on whether the `mapreduce.application.framework.path` is set or not, the MR framework will use different config to construct the classpath